### PR TITLE
 FeedbackQuestionDetails abstract class instance variables should be 'protected' #5434 

### LINKED ISSUE #5434: FeedbackQuestionDetails abstract class instance variables should be 'protected'
Currently they are set to public visibility

### LINKED ISSUE #5435: Instance variables of all subclasses of FeedbackQuestionDetails should have 'private' visibility
Currently most of them are set to 'public'

### PR DESCRIPTION
**Pull request by rclakmal**
Fixes #5434 
Fixes #5435

### CODE REVIEW

**Comment by YongJieYongJie** [comment]
For discussion:

 - Maybe we should not expose setters to all the instance variables, because
  - some classes have constructors for various kinds of instantiation or methods to set groups of instance variables all at once, and we should make people use these
  - encapsulating everything but automatically generating setters for all instance variables can be more misleading than not encapsulating at all

**Comment by damithc** [comment]
Yes, we can omit setters that are not in use.

**Comment by junhaoyap** [comment]
1. What ^ said, we can omit unused setters and getters, when future maintainers need them, let them add it

**Comment by junhaoyap** [comment]
2. There are multiple unnecessary uses of the keyword ```this```, http://stackoverflow.com/questions/2411270/when-should-i-use-this-in-a-class, let's clean up some of them while we are at this PR as it is very confusing for ```this``` to be used so freely

File: src/main/java/teammates/common/datatransfer/FeedbackConstantSumQuestionDetails.java [hunk 1]
```diff
@@ -311,14 +311,14 @@ public String getQuestionAdditionalInfoHtml(int questionNumber,
         String optionFragmentTemplate = FeedbackQuestionFormTemplates.MSQ_ADDITIONAL_INFO_FRAGMENT;
         StringBuilder additionalInfo = new StringBuilder();
 
-        if (this.distributeToRecipients) {
+        if (this.isDistributeToRecipients()) {
```
Comment (junhaoyap) [inline src/main/java/teammates/common/datatransfer/FeedbackConstantSumQuestionDetails.java]: For instance here, the scope is not ambiguous at all

**Comment by junhaoyap** [comment]
@rclakmal overall looks good with mostly changes to using getters and setters, please address the above 2 issues and we'll look at it again :)

**Comment by rclakmal** [comment]
@junhaoyap @damithc @YongJieYongJie Thanks for the review. Included only utilized gets/sets methods.

@junhaoyap shall we create a separate issue to remove the unnecessary `this` usage ? WDYT ?

**Comment by rclakmal** [comment]
@junhaoyap I removed the unnecessary this usage from FQD and all its sub classes. Let me know if you want to keep it here or do it via another issue.

**Comment by damithc** [comment]
@junhaoyap review pending

**Comment by junhaoyap** [comment]
Looks good - let's keep it here for now, extremely unlikely for this to get reverted considering the scope of the PR, though this might signal that we have a lot of unnecessary ```this``` sprinkled around in the rest of the project

**Comment by junhaoyap** [comment]
@damithc please take a look

**Comment by rclakmal** [comment]
Fixed merge conflicts

File: src/test/java/teammates/test/cases/ui/StudentProfileEditSaveActionTest.java [hunk 2]
```diff
@@ -1,146 +1,146 @@
...
+        spa.moreInfo = StringHelper.trimIfNotNull(submissionParams[11]);
+        spa.modifiedDate = null;
+        
+        return spa;
+    }
+
+    private StudentProfileEditSaveAction getAction(String[] submissionParams) {
+        return (StudentProfileEditSaveAction) gaeSimulation.getActionObject(uri, submissionParams);
+    }
+
```
Comment (damithc) [inline src/test/java/teammates/test/cases/ui/StudentProfileEditSaveActionTest.java]: Any idea why this entire block is showing up in diff?

**Comment by damithc** [comment]
Waiting for clarification

**Comment by rclakmal** [comment]
@damithc I believe it had something to with tabs/spaces. I reverted it to previous version. Somehow didn't notice it and slipped during  the last commit :-(

**Pull request merged**